src/recovery.rs

```diff
@@ -103,15 +117,10 @@ pub struct Recovery {
 
     pub lost_count: usize,
 
-    bytes_in_flight: usize,
-
-    cwnd: usize,
-
-    recovery_start_time: Option<Instant>,
-
-    ssthresh: usize,
-
     pub loss_probes: [usize; packet::EPOCH_COUNT],
+
+    // Congestion Control
+    pub cc: Option<Box<dyn cc::CongestionControlTrait>>,
```

Comment:
I don't think this needs to be an `Option`. Instead we can just turn the `Recovery::default()` constructor into `Recovery::new(cc: CongestionControlID)` (or whatever the enum used for selecting a CC in `Config` is), and create the object when the `Recovery` object is also created. This also means that this field wouldn't need to be public, and the whole `OptionExt` thing can go away too.

Reply:
I was wondering too... let me try.

Reply:
Now `Option<Box<dyn cc::CongestionControlTrait>>` -> `Box<dyn cc::CongestionControlTrait>` and all helper method are removed.

Reply:
As per my previous comment, this also doesn't need to be public, by making the `Recovery` consructor take the CC algorithm and create the correct one, rather than create a default CC state, and then change it later.

Reply:
So currently we don't have a constructor of Recovery other than default() which doesn't accept a paramter, so I changed Default implementation into a new() constructor in Recovery which accept Config as a paramter. In that way we can use Config object other than CC initilization going forward, but let me know this approach is ok.